nshlib/nsh_parse.c

```diff
@@ -59,11 +59,29 @@
 #  define HAVE_MEMLIST 1
 #endif
 
+/* If CONFIG_NSH_ALIAS is enabled, the alias strings might need dynamic
+ * memory, in case the alias has arguments and is set like:
+ *
+ * $ alias ls='ls -l'
+ *
+ * In this case the command verb and the arguments need to be separated, much
+ * like the argument separation is done with environment variable expansion.
+ *
+ * This needs a new working buffer in order to keep the original alias string
+ * intact.
+ */
+
+#ifdef CONFIG_NSH_ALIAS
+#  define ALIAS_ALLOCS           2
+#else
+#  define ALIAS_ALLOCS           0
+#endif
+
 #if defined(HAVE_MEMLIST) && !defined(CONFIG_NSH_MAXALLOCS)
 #  ifdef CONFIG_NSH_ARGCAT
-#    define CONFIG_NSH_MAXALLOCS (2*CONFIG_NSH_MAXARGUMENTS)
+#    define CONFIG_NSH_MAXALLOCS ((2*CONFIG_NSH_MAXARGUMENTS) + ALIAS_ALLOCS)
```

Comment:
```suggestion
#    define CONFIG_NSH_MAXALLOCS ((2 * CONFIG_NSH_MAXARGUMENTS) + ALIAS_ALLOCS)
```

Reply:
I like the suggestion but the reason I did not add the spaces originally was that it would exceed the 78 character limit. 

```
#    define CONFIG_NSH_MAXALLOCS ((2 * CONFIG_NSH_MAXARGUMENTS) + \ 
                                                                  ALIAS_ALLOCS)
```

So it is a trade-off, add spaces to improve readability but add extra white space, or keep it like it is.

The original code did not have the whitespaces so that is why I did not add them + the line change myself.

What do you think, which is better ? I'm fine either way.

Reply:
I approved the PR, so I'm fine with current variant. If adding spaces makes macro to become multiline then I think better to keep as is.